components/hubspot/hubspot.app.mjs

```diff
@@ -204,41 +223,43 @@ export default {
         startDate: Date.now(),
         endDate,
       };
-      return await this.makeAxiosRequest("GET", "/calendar/v1/events/task", params);
+      return this.makeAxiosRequest("GET", "/calendar/v1/events/task", params);
     },
     async getContactProperties() {
-      return await this.makeAxiosRequest("GET", "/properties/v1/contacts/properties");
+      return this.makeAxiosRequest("GET", "/properties/v1/contacts/properties");
     },
     async createPropertiesArray() {
       const allProperties = await this.getContactProperties();
       return allProperties.map((property) => property.name);
     },
     async getDealProperties() {
-      return await this.makeAxiosRequest("GET", "/properties/v1/deals/properties");
+      return this.makeAxiosRequest("GET", "/properties/v1/deals/properties");
     },
-    async getDealStages() {
-      return await this.makeAxiosRequest("GET", "/crm-pipelines/v1/pipelines/deal");
+    async getDealStages(pipelineId) {
+      const pipelines = await this.getPipelines("deal");
+      const pipeline = pipelines.filter((pipeline) => pipeline.id == pipelineId);
+      return pipeline[0].stages;
```

Comment:
Since you are always looking for the first occurrence, In my view you could use `find`

`const pipeline = pipelines.find((pipeline) => pipeline.id == pipelineId);`
`return pipeline?.stages;`

I suggest you add the `?` just in case, if the `find` return `null`